argh_derive/src/lib.rs

```diff
@@ -506,9 +489,14 @@ fn unwrap_fields<'a>(fields: &'a [StructField<'a>]) -> impl Iterator<Item = Toke
                 Optionality::Optional | Optionality::Repeating => {
                     quote! { #field_name: #field_name.slot }
                 }
-                Optionality::Defaulted(tokens) => {
+                Optionality::Defaulted(string) => {
                     quote! {
-                        #field_name: #field_name.slot.unwrap_or_else(|| #tokens)
+                        #field_name: {
+                            let parse_func = #field_name.parse_func;
+                            #field_name.slot.unwrap_or_else(|| {
+                                (parse_func)(#string).expect("Parsing default attribute")
+                            })
+                        }
```

Comment:
Here I call a `parse_func` on `default` value.
`parse_func` must be borrowed from the field before using in closure because the error happens "cannot move the value because it is already borrowed in closure".

So if the option doesn't exist and parsing fails then developer will see an error in runtime (yeah) like "Parsing default attribute: expected integer but string literal is found" (this is not real message)